@xen-orchestra/lite/src/libs/xen-api/operations/vm-operations.ts

```diff
@@ -100,8 +103,8 @@ export function createVmOperations(xenApi: XenApi) {
       )
     },
 
-    setAffinityHost: (vmRefs: XenApiVm['$ref'], hostRef: XenApiHost['$ref'] | null) =>
-      Promise.all(toArray(vmRefs).map(vmRef => xenApi.call('VM.set_affinity', [vmRef, hostRef ?? '']))),
+    setAffinityHost: (vmRefs: VmRefs, hostRef: HostRef | undefined) =>
+      Promise.all(toArray(vmRefs).map(vmRef => xenApi.call('VM.set_affinity', [vmRef, hostRef ?? OPAQUE_REF.EMPTY]))),
```

Comment:
IMO, It would be "better" to be explicit here.
`vm.setAffinityHost([vmRef], hostRef)` // it will remove the affinity host if hostRef is undefined, and it maybe can lead to unwanted behavior
`vm.setAffinityHost([vmRef], OPAQUE_REF.EMPTY)` // we explicitly pass an empty ref.
Or maybe simply use `null` value?
`vm.setAffinityHost([vmRef], null)` // we explicitly remove the affinity host
```suggestion
    setAffinityHost: (vmRefs: VmRefs, hostRef: HostRef | OPAQUE_REF.EMPTY) =>
      Promise.all(toArray(vmRefs).map(vmRef => xenApi.call('VM.set_affinity', [vmRef, hostRef]))),
```